feat: add `stats/incr/nanmidrange`

jalajk3004 opened the pull request:
Resolves #5578

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a package nanmidrange, which works like midrange in which it calculates midrange incrementally ignoring NaN values

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5578

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

jalajk3004 committed:
edit

jalajk3004 committed:
edit

jalajk3004 committed:
Merge branch 'develop' of https://github.com/jalajk3004/stdlib into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
Merge branch 'stdlib-js:develop' into develop

jalajk3004 committed:
added nanmidrange

jalajk3004 committed:
error

jalajk3004 committed:
lint

jalajk3004 committed:
lint

jalajk3004 committed:
changes in index

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmidrange/docs/types/Test.ts:
```diff
@@ -0,0 +1,63 @@
+/*
```
This file should be named `test.ts` (lowercase), not `Test.ts`. The stdlib convention is to use only lowercase letters, numbers, and underscores in filenames. If you look at the reference `@stdlib/stats/incr/midrange/docs/types/`, the file is `test.ts`.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmidrange/docs/types/index.d.ts:
```diff
@@ -0,0 +1,64 @@
+* ## Notes
+*
+* -   The mid-range is the arithmetic mean of maximum and minimum values. Accordingly, the mid-range is the midpoint of the range and a measure of central tendency.
+* -   If provided `NaN` or a value which, when used in computations, results in `NaN`, the accumulated value is `NaN` for all future invocations.
```
This note was carried over from the non-nan `incrmidrange` type definition, but it's incorrect here. The whole point of `nanmidrange` is that NaN values are **ignored**, not propagated. This line tells users the exact opposite of what the function does.

We should remove this note entirely.

Planeshifter replied:
Remove the Notes block (lines 26-29). The NaN-poisoning note is copied from `incrmidrange` and is incorrect here, as `incrnanmidrange` ignores NaN values rather than propagating them.

Planeshifter reviewed (changes requested):
Thanks for your PR! I left some review comments; better later than never, I thought.

Planeshifter commented on lib/node_modules/@stdlib/stats/incr/nanmidrange/package.json:
```diff
@@ -0,0 +1,71 @@
+{
+  "name": "@stdlib/stats/incr/nanmidrange",
+  "version": "0.0.0",
+  "description": "Compute a mid-range incrementally, ignoring `NaN` values.",
```
In `package.json` descriptions, `NaN` should not be wrapped in backticks.

```suggestion
  "description": "Compute a mid-range incrementally, ignoring NaN values.",
```

Planeshifter committed:
fix: address review feedback in stats/incr/nanmidrange

- Rename docs/types/Test.ts to test.ts so the TypeScript type test is picked up on case-sensitive filesystems
- Remove the incorrect NaN-propagation note from index.d.ts (the accumulator ignores NaN)
- Correct the accumulator JSDoc @returns type to (number|null)
- Drop backticks from package.json description; add nan/ignore keywords
- Simplify examples output and guard against a null accumulated value
- Bump copyright year to 2026

Planeshifter reviewed (approved):
Let's land it.
